serverless/functions/serverless-functions-on-cluster-builds.adoc

```diff
@@ -0,0 +1,92 @@
+:_content-type: PROCEDURE
+[id="serverless-functions-on-cluster-builds"]
+= Building and deploying functions on-cluster
+:context: serverless-functions-on-cluster-builds
+include::_attributes/common-attributes.adoc[]
+
+toc::[]
+
+You can use the Knative (`kn`) CLI to push a function project directly to the cluster, which triggers a build and deployment of the function. You can also push a function project to a source code repository branch, which triggers a build of the function with the updated changes and deploys the updated function on the cluster.
+
+:FeatureName: {FunctionsProductName}
+include::snippets/technology-preview.adoc[leveloffset=+1]
+
+[id="prerequisites_serverless-functions-on-cluster-builds"]
+== Prerequisites
+
+* xref:../../cicd/pipelines/installing-pipelines.adoc[{pipelines-title}] must be installed on your cluster.
+
+* You have installed the OpenShift (`oc`) CLI.
+
+* You have installed the Knative (`kn`) CLI.
+
+[id="serverless-functions-on-cluster-builds-procedure"]
+== Procedure
+
+. In each namespace where you want to run {pipelines-shortname} and deploy a function, you must create the following resources:
+
+.. Create the functions buildpacks Tekton task to be able to build the function image:
++
+[source,terminal]
+----
+$ oc apply -f https://raw.githubusercontent.com/openshift-knative/kn-plugin-func/tree/serverless-1.22.0/pipelines/resources/tekton/task/func-buildpacks/0.1/func-buildpacks.yaml
+----
+
+.. Create the `kn func` deploy Tekton task to be able to deploy the function in the pipeline:
++
+[source,terminal]
+----
+$ oc apply -f https://raw.githubusercontent.com/openshift-knative/kn-plugin-func/tree/serverless-1.22.0/pipelines/resources/tekton/task/func-deploy/0.1/func-deploy.yaml
+----
+
+. Create a function:
++
+[source,terminal]
+----
+$ kn func create <function_name> -l <runtime>
+----
+
+. Get a reference to the remote Git repository that hosts your function source code.
+
+. Initialize a Git repository in your function project and add a reference to the remote repository.
+
+. Update the configuration in the `func.yaml` file for your function project to enable on cluster builds for the Git repository:
```

Comment:
These steps are all a little too vague or incomplete, if we want to be thorough. After creating the function in the first step, the user needs to perform steps similar to the following.

* Initialize the project locally as a git repo
* Create a new repo in a tool such as github
* Use the commands provided by github UI to associate the project with the github repository
* Add the current project files to the repo with `git add`
* Commit that change with `git commit`
* Push those changes to github with `git push`

Then they can perform the on-cluster build.

It's sort of complicated and gets into areas that are really not ours to document. I wonder if instead of providing bulleted step by step procedures here, we use prose to describe the concepts. Would it be clearer and less prone to error?

> After creating a new function project with a command similar to that shown above, you'll need to add the project to a git repository and ensure that repository is available to the cluster. Then update `func.yaml` with your repository information as shown below.

Or something similar.

Reply:
Yeah I think that might be a good way to go